### PR TITLE
Fix issue #134

### PR DESCRIPTION
**Pull request by switch-cheehau**
Solve issue #134

Please review.

### CODE REVIEW

**Comment by aimeos** [comment]
Thank you for your PR!
Can you please use tabs instead of spaces for indenting like we do in the rest of the code?

File: client/html/themes/aimeos.js [hunk 1]
```diff
@@ -1225,14 +1225,28 @@ AimeosCatalog = {
...
 				}
 			}
+
+            if(!AimeosCatalog.validateVariant()) {
+                $(".addbasket .btn-action", parent).addClass("btn-disabled").attr("disabled", "disabled");
+            }
+
+            if(!AimeosCatalog.validateVariant()) {
+                $(".articleitem").removeClass("stock-actual");
+            }
```
Comment (aimeos) [inline client/html/themes/aimeos.js:1244]: Those two statements can be combined

File: client/html/themes/aimeos.js [hunk 2]
```diff
@@ -1225,14 +1225,28 @@ AimeosCatalog = {
...
 		});
+
+        if(!AimeosCatalog.validateVariant()) {
+            $(".articleitem").removeClass("stock-actual");
+        }
```
Comment (aimeos) [inline client/html/themes/aimeos.js:1249]: Why are these lines necessary? They are executed on page load

Comment (switch-cheehau) [reply client/html/themes/aimeos.js:1249]: These lines is to hide the message if no attribute selected.
image

Hope it make sense.
Please guide me to do proper way.

File: client/html/themes/aimeos.js (hunk 1 above)
Comment (switch-cheehau) [reply client/html/themes/aimeos.js]: fix at ffacf7d

File: client/html/themes/aimeos.js (hunk 2 above)
Comment (aimeos) [reply client/html/themes/aimeos.js]: Maybe it's better to remove the `stock-actual` class by default for `select` and `group` products in the template:
https://github.com/aimeos/ai-client-html/blob/master/client/html/templates/catalog/detail/body-standard.php#L270

Otherwise, customer will notice that the line disappears because the JS is executed after the page is loaded. Thus, we would suggest something like:

```
<div class="articleitem <?= !in_array( $this->detailProductItem->getType(), ['select', 'group'] ) ? 'stock-actual' : '' ?>"
```
The same should be done here:
- https://github.com/aimeos/ai-client-html/blob/master/client/html/templates/catalog/lists/items-body-list.php#L96
- https://github.com/aimeos/ai-client-html/blob/master/client/html/templates/common/partials/products-standard.php#L238

**Comment by aimeos** [comment]
Looks great now and thanks a lot! :-)

File: client/html/templates/catalog/lists/items-body-list.php [hunk 3]
```diff
@@ -176,7 +176,7 @@
 										required="required" <?= !$productItem->isAvailable() ? 'disabled' : '' ?>
 										name="<?= $enc->attr( $this->formparam( array( 'b_prod', 0, 'quantity' ) ) ) ?>"
 									><!--
-									--><button class="btn btn-primary" type="submit" value="">
+									--><button class="btn btn-primary btn-action" type="submit" value="" <?= !$productItem->isAvailable() ? 'disabled' : '' ?>>
```
Comment (switch-cheehau) [inline client/html/templates/catalog/lists/items-body-list.php:179]: i noticed <kbd>$productItem->isAvailable()</kbd> does not function properly.


image

Even the stock quantity is zero, it still return true

**Comment by aimeos** [comment]
`isAvailable()` doesn't check for stock levels as these data is loaded async via XHR because they are highly volatile. You can find the stock JS code for modifying the HTML here:
https://github.com/aimeos/ai-client-html/blob/master/client/html/templates/catalog/stock/body-standard.php
